New website

marienfressinaud opened the pull request:
I'm working on a new website for the anniversary of FreshRSS (see https://github.com/FreshRSS/FreshRSS/discussions/4294).

**The development website can be found at [next.freshrss.org](https://next.freshrss.org)**.

Note that for this redesign I used [my own static site generator, Boop!](https://framagit.org/marienfressinaud/boop) which is a lot simpler than Pelican. Boop! is finished so it's very stable. It is also very simple, but I think it's good enough for a site like freshrss.org. And if for any reason we decide to choose a different tool, it should be very easy to switch :)

Checklist:

- [x] create a dedicated page for the 10 years
- [x] add a link to the instructions on how to install FRSS
- [x] add a "legal" page (including credits for illustrations, font, etc.)
- [x] use the OpenSans font
- [x] add license to the repo and a link in the legal page
- [x] add a "go to top" button
- [x] add a page with a list of existing instances (see https://github.com/FreshRSS/FreshRSS/blob/edge/docs/en/admins/14_CloudProviders.md)
- [x] add a page to contact us
- [x] create a template for blog.html and articles (they are no longer accessible via the home page, but the pages still exist to not break links)
- [x] setup OpenGraph tags properly (mostly a page illustration to setup and a description to add to pages)
- [x] check accessibility (colors, keyboard navigation, screen readers, images alt, etc.)
- [x] automate deployment on the server
- [x] ask for testimonials (~~and add a page dedicated to testimonials?~~)
- [x] ~~add a page to list events (?)~~
- [x] ~~add a page with more screenshots (?)~~
- [x] Optimize the website (https://pagespeed.web.dev)
- [x] Create a Mobilizon event (and/or a pad) to organize the event at Grenoble and change the link of "Learn about the event" on the [10 years](https://next.freshrss.org/10-years.html) page
- [x] fix typos and frenglish (I'll need help for this one :))
- [x] (1st October) Open [a discussion](https://github.com/FreshRSS/FreshRSS/discussions) named "Celebrating 10 years" and change the link of "Express yourself" on the 10 years page
- [x] Edit the "deploy" workflow to deploy the main branch

math-GH commented:
Is there any reason to use the very old jQuery 2.1.1?

marienfressinaud commented:
There's no reasons at all, that's why this file is removed :p

math-GH commented:
> There's no reasons at all, that's why this file is removed :p

oh, sorry. I miss understood it. I have overseen the remove icon

Alkarex commented on templates/page.html:
```diff
@@ -0,0 +1,92 @@
+                <p>
+                    <small>
+                        FreshRSS does not meet your expectations?<br>
+                        <a href="https://alternativeto.net/software/FreshRSS">Have a look at some alternatives.</a>
```
Or vote for FreshRSS there ;-)

marienfressinaud replied:
Fixed :)

marienfressinaud commented:
I deployed the actual version of the website at https://next.freshrss.org. It's still work in progress and few things are not working yet, but it gives a better idea of how it looks :)

Frenzie commented:
Join the chat links to some signup page instead of to a room?

aledeg commented on pages/10-years.html:
```diff
@@ -0,0 +1,75 @@
+
+            <p>
+                The people behind FreshRSS wants their users to meet during
+                October. You can even help us by organizing an event in your
```
Maybe `own` can be removed

marienfressinaud replied:
Done :+1:

Frenzie commented on pages/10-years.html:
```diff
@@ -0,0 +1,75 @@
+            </h3>
+
+            <p>
+                The people behind FreshRSS wants their users to meet during
```
```suggestion
                The people behind FreshRSS want their users to meet during
```

marienfressinaud replied:
Done :+1:

math-GH commented:
Nice!

https://pagespeed.web.dev/report?url=https%3A%2F%2Fnext.freshrss.org%2F&hl=en
![grafik](https://user-images.githubusercontent.com/1645099/187762037-c8b6e6b0-e1d3-41f0-b32d-86225fe67a47.png)

marienfressinaud commented:
> Join the chat links to some signup page instead of to a room?

@Frenzie It's an invitation link: we have to give it otherwise people will not be able to join the team (it's private and we can't change it). Btw, the team is limited to a certain number of users (I think it's 250), so I'm afraid that we reach this limit some day :/

> Nice! https://pagespeed.web.dev/report?url=https%3A%2F%2Fnext.freshrss.org%2F&hl=en

@math-GH and I didn't start to optimize anything, it's just my basic Web server configuration + a simple static website :)

marienfressinaud commented:
I wasn't happy with the website entirely using the Open Sans font. I kept it for titles, but I chose PT Serif for the text… but I'm still not satisfied :sweat_smile: Maybe I'll stay with a sans-serif font? (I'll try another day)

I also added a simple contact page: https://next.freshrss.org/contact.html

Frenzie commented:
@marienfressinaud 
> @Frenzie It's an invitation link: we have to give it otherwise people will not be able to join the team (it's private and we can't change it). Btw, the team is limited to a certain number of users (I think it's 250), so I'm afraid that we reach this limit some day :/

I suppose that makes sense the first time but that's very confusing if you're already registered. :-/

marienfressinaud commented:
> I suppose that makes sense the first time but that's very confusing if you're already registered. :-/

I'm not sure to understand what's confusing: the link redirects to the FreshRSS team on Mattermost/Framateam, which is why I expect. Either way, we don't have a choice here, it's how Mattermost works.

---

I almost finished what I wanted to do (in particular, the website is automatically deployed :)). The three last things:

- about the testimonials, I wonder if we already have a list somewhere? (maybe we could find some in articles that mentions FreshRSS? and on Twitter as well)
- about the events: I'll do something at Grenoble (France) in October (more on this later!), but while we don't have at least 2 more events I think a dedicated page is too much
- about a dedicated screenshots page: I'll let this for the future. I think the website is good enough like this (and I'm very not motivated to take screenshots :sweat_smile: )

Frenzie commented:
> I'm not sure to understand what's confusing: the link redirects to the FreshRSS team on Mattermost/Framateam, which is why I expect. Either way, we don't have a choice here, it's how Mattermost works.

If you're not logged in, going to https://framateam.org/freshrss/channels/town-square looks like this:

<img src=https://user-images.githubusercontent.com/202757/188826763-9e907912-46fd-4bd4-ab7b-f72680736bb6.png width=40%>

And there's a nice "create an account" in the corner.

The other link gives you the opposite:

<img src=https://user-images.githubusercontent.com/202757/188827035-e338cb69-a95c-4f87-9b4d-ca672f763483.png width=40%>

The experience with https://framateam.org/freshrss/channels/town-square is significantly better.

1. It's a "real" link, unlike https://framateam.org/signup_user_complete/?id=e2680d3e3128b9fac8fdb3003b0024ee
2. Any site that emphasizes creating accounts over logging in is horrendously annoying.
3. It looks like you were mistaken about <https://framateam.org/freshrss/channels/town-square>? :-)

marienfressinaud commented:
Aaah, right, it happens when you're not logged in :sweat_smile: It seems I found the solution: the `id` parameter can be put at the end of https://framateam.org/freshrss/channels/town-square and it allows to join the team as well :)

Alkarex commented on index.html:
```diff
@@ -0,0 +1,286 @@
+            </h3>
+
+            <p>
+                Follow your feeds with or without third-party application.
```
```suggestion
                Follow your feeds with or without [third-party application](https://github.com/FreshRSS/FreshRSS/blob/edge/README.md#apis--native-apps).
```

Frenzie replied:
(applied a small edit to @Alkarex' suggestion)

Alkarex commented on index.html:
```diff
@@ -0,0 +1,286 @@
+            </h3>
+
+            <p>
+                FreshRSS can manage +100k articles without complaining.
```
Actually 1M+ articles and 50k+ feeds :-)
The large number of feeds is a [recent feature](https://github.com/FreshRSS/FreshRSS/pull/4347).

Alkarex commented:
Excellent work @marienfressinaud 👍🏻

math-GH commented:
I would like to have a smaller FreshRSS logo.

Before:
![grafik](https://user-images.githubusercontent.com/1645099/191309873-3869e743-065b-4d53-b72e-2312624a6c52.png)

Suggestion: width = 250px
![grafik](https://user-images.githubusercontent.com/1645099/191310007-fc6b3d19-8127-41e2-a787-e4ef1ec46590.png)

Mobile:

Before:
![grafik](https://user-images.githubusercontent.com/1645099/191310906-a5cd69d2-da27-447d-9d0d-d2c542503254.png)

Suggestion: mobile view with max-width = 75%
![grafik](https://user-images.githubusercontent.com/1645099/191310486-3aac20b0-775d-42d2-a862-421e97382d02.png)

math-GH commented:
Little speed ups:
Analyzed with https://pagespeed.web.dev/report?url=https%3A%2F%2Fnext.freshrss.org%2F&form_factor=mobile

1. Set `width` and `height` for images
2. use jpeg or webp for the images, especially the screenshot.png (webp: https://caniuse.com/?search=webp)

Frenzie commented:
Also see https://pngquant.org/ (likely to be visually identical and smaller than WebP)

marienfressinaud commented:
I optimized the images (png → webp) and added cache on wepb and svg images. I also changed the logo by a SVG image. PageSpeed still complains about the screenshot because it could be resized, but if I resize it to 1000x563px, it actually increases the size of the file (I think I should retake a screenshot to the wanted sizes). I think I'm totally done with this :) (except the "express yourself" link on the 10 years page that I'll change the day of the deployment)

math-GH commented:
This tool compress webp files

https://tinypng.com/

(I cannot send webp files via comment)

marienfressinaud commented:
I already use [Yoga](https://yoga.flozz.org/) to compress files. In fact both TinyPNG and Yoga can reduce the size of the files by reducing drastically the quality. I'd like to keep a screenshot of good quality. Actually the screenshot is 45 kB, which I think is small enough for a screenshot (it can be compared to the screenshot of 217 kB on the actual homepage ^^) The few kB that we could save are really not worth the time we can spend on it.

marienfressinaud commented:
Let's merge and deploy in production!
